gnocchi/utils.py

```diff
@@ -133,19 +133,17 @@ def timestamp_to_datetime(v):
         v.astype(float) / 10e8).replace(tzinfo=iso8601.iso8601.UTC)
 
 
-def to_timespan(value):
+def to_timespan(value, allow_le_zero=False):
```

Comment:
what's allow_le_zero for?

Reply:
allow lesser or equal than zero. It's used on L106.

Reply:
right, i'm just wondering where we actually allow that currently? becauses the previous code does not support this condition

Reply:
Oh it does support it, see line 106 again. When converting "-5 seconds" to a *datetime64*, what's computed is `now() - 5 seconds`. So we need to allow for those negative timespan.

The current code used `pandas.to_timedelta` in `to_timestamps` where this new one use the new `to_timespan` with `allow_le_zero` set to `True`.

Welcome back ;)

Reply:
oh word.